libs/eavmlib/src/uart.erl

```diff
@@ -19,24 +19,141 @@
 %
 
 -module(uart).
--export([open/1, open/2, close/1, read/1, write/2]).
+-export([open/1, open/2, close/1, read/1, read/2, write/2]).
 
+-type peripheral() :: string() | binary().
+% The peripheral `Name' may be one of: `"UART0"' | `"UART1"' | `"UART2"' | `<<"UART0">>' | `<<"UART1">>' | `<<"UART2">>'.
+
+-type uart_opts() :: [
+    {tx, _Tx_pin :: integer()}
+    | {rx, _Rx_pin :: integer()}
+    | {rts, _Rts_pin :: integer()}
+    | {cts, _Cts_pin :: integer()}
+    | {speed, _Speed :: pos_integer()}
+    | {data_bits, 5..8}
+    | {stop_bits, 1 | 2}
+    | {event_queue_len, _Qlen :: pos_integer()}
```

Comment:
Is there a reason for the underscore prefix?

Reply:
I thought I was getting some warnings either during compile, from dialyzer, or gradualizer, but I can't reproduce that now. I will remove them.